extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -510,11 +510,13 @@ public Integer get() {
     }
 
     public ThreadFactory createThreadFactory() {
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
         AtomicInteger threadCount = new AtomicInteger(0);
         return runnable -> {
             VertxThread thread = VertxThreadFactory.INSTANCE.newVertxThread(runnable,
                     "executor-thread-" + threadCount.getAndIncrement(), true, 0, null);
             thread.setDaemon(true);
+            thread.setContextClassLoader(tccl);
```

Comment:
@stuartwdouglas this change doesn't look related? Is it something you want in?

Reply:
oops, fixed